securedrop/tests/conftest.py

```diff
@@ -110,32 +115,41 @@ def insecure_scrypt() -> Generator[None, None, None]:
 
 
 @pytest.fixture(scope="session")
-def gpg_key_dir() -> Generator[Path, None, None]:
-    """Set up the journalist test key in GPG and the parent folder.
+def setup_journalist_key_and_gpg_folder(
+    tmp_path_factory: TempPathFactory
+) -> Generator[Tuple[str, Path], None, None]:
+    """Set up the journalist test key and the key folder, and reduce source key length for speed.
 
     This fixture takes about 2s to complete hence we use the "session" scope to only run it once.
     """
-    with TemporaryDirectory() as tmp_gpg_dir_name:
-        tmp_gpg_dir = Path(tmp_gpg_dir_name)
+    tmp_gpg_dir = tmp_path_factory.mktemp("gpg_dir")
 
-        # GPG 2.1+ requires gpg-agent, see #4013
-        gpg_agent_config = tmp_gpg_dir / "gpg-agent.conf"
-        gpg_agent_config.write_text("allow-loopback-pinentry")
+    # GPG 2.1+ requires gpg-agent, see #4013
+    gpg_agent_config = tmp_gpg_dir / "gpg-agent.conf"
+    gpg_agent_config.write_text("allow-loopback-pinentry\ndefault-cache-ttl 0")
```

Comment:
Important change here: as I understand it, `default-cache-ttl 0` disables the caching of the GPG passphrases. 

This was needed because previously, the source's GPG passphrase was not required to call decryption/encryption operations if that passphrase was already in the cache.

I've put a comment below in `test_encrypt_and_decrypt_journalist_reply()` in `test_encryption.py` showing an example where a source would be able to decrypt using another source's key, thanks to passphrase caching.

We should discuss whether we want to keep `default-cache-ttl 0`, and if so how to enable it when running the app (ie. outside of the test suite).